src/functions_framework/__init__.py

```diff
@@ -75,30 +97,47 @@ def _is_binary_cloud_event(request):
         and request.headers.get("ce-id")
     )
 
+# TODO(joelgerard): is this really legacy as I have stated?
+def _run_binary_legacy_cloud_event(function, request):
+    # Support CloudEvents in binary content mode, with data being the
+    # whole request body and context attributes retrieved from request
+    # headers.
+    data = request.get_data()
+    context = Context(
+        eventId=request.headers.get("ce-eventId"),
+        timestamp=request.headers.get("ce-timestamp"),
+        eventType=request.headers.get("ce-eventType"),
+        resource=request.headers.get("ce-resource"),
+    )
+    function(data, context)
+
+def _run_legacy_cloud_event(function, request):
```

Comment:
We don't need to handle legacy events. Remove this.

Reply:
OK. Why are we not worried about backwards compatibility?

Reply:
The existing logic was taken directly from the (internal) invoker for the Python 3.7 runtime. I would be surprised if it wasn't being used there.